src/update_handler.c

```diff
@@ -366,6 +373,15 @@ static gboolean casync_extract(RaucImage *image, gchar *dest, int out_fd, const
 		goto out;
 	}
 
+	res =  file_to_process_stdin(image->filename, sproc, &ierror);
```

Comment:
Does casync/desync read the input index synchronized with writing the output image? Otherwise this wouldn't help much as the index is usually very small.